Core/Src/main.c

```diff
@@ -26,5 +42,26 @@ int main()
 void UART1_RxCallback(void) 
 {
   GPIO_TogglePin(LED_GREEN_Port, LED_GREEN_Pin);
-  UART1_ReceiveIT(buf, 1);
+  UART1_ReceiveIT(bufUART, 1);
+
+  if(bufUART[0] == 'd')
+  {
+    DS3231_Date_t date;
+    date.Year = 21;
+    date.Month = 1;
+    date.Date = 14;
+    date.Day = 3;
+
+    DS3231_SetDate(&DS3231, &date);
+  }
+  else if (bufUART[0] == 't')
+  {
+    DS3231_Time_t time;
+    time.Hours = 0;
+    time.Minutes = 3;
+    time.Seconds = 50;
+
+    DS3231_SetTime(&DS3231, &time);
+  }
+  
 }
```

Comment:
Add a newline at the end of the file here and in other places where the newline is missing